### ISSUE TITLE
Error inside a policy is misreported as PolicyWonk.Enforce.PolicyError

### ISSUE THREAD

**Issue by wasnotrice** [boydm/policy_wonk#3]
Hi, thanks for policy_wonk!

I ran into a strange situation where I created a policy that had an error inside it, but when the `PolicyWonk.Enforce` plug fired for this policy, policy_wonk caught my error and mistakenly reported it as a `PolicyWonk.Enforce.PolicyError` like this: 

```
** (PolicyWonk.Enforce.PolicyError) Unable find to a policy definition for:
Policy: :policy_with_error
In any of the following modules...
MyApp.MyController
MyApp.Policies
```

Here's an simple policy that causes the issue:

```elixir
defmodule MyApp.Policies do
  use PolicyWonk.Enforce
  @behaviour PolicyWonk.Policy

  def policy(assigns, :policy_with_error) do
    # Call a function with the wrong parameters
    IO.inspect(:some_value, "an incorrect argument")
  end
end
```

Since `PolicyWonk.Utils.call_down_list/2` catches any `FunctionClauseError`, it thinks that the error raised by the bad call to `IO.inspect/2` means that there was an error calling `MyApp.Policies.policy/2`.

I'd be happy to submit a PR to fix this—I think that checking the module/function/arity on the rescued error should take care of it.

Cheers!

**Comment by boydm** [comment]
Hey. Good catch. I didn't think of that.

Go ahead and spin up a PR if you like. That would speed things along. Or I could get to it maybe over the weekend.

**Comment by boydm** [comment]
Very helpful PR. I've merged it, and published the update to hex as version 0.1.3
